From 82069a7a02632aa60fa5c69415bf891ede7d6fd4 Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Tue, 5 Dec 2023 16:55:17 +0000
Subject: [PATCH] dts: bcm2712: put usb under /axi not /soc

On 2712, the DWC USB controller is no longer attached to the Videocore
30-bit bus with its associated aliases, and can see the bottom 4GB of
RAM directly.

Ideally it should make use of IOMMU6 but for now software bounce buffers
get it working.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2712.dtsi | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

--- a/arch/arm/boot/dts/bcm2712.dtsi
+++ b/arch/arm/boot/dts/bcm2712.dtsi
@@ -133,19 +133,6 @@
 			status = "disabled";
 		};
 
-		usb: usb@7c480000 {
-			compatible = "brcm,bcm2835-usb";
-			reg = <0x7c480000 0x10000>;
-			interrupts = <GIC_SPI 73 IRQ_TYPE_LEVEL_HIGH>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-			clocks = <&clk_usb>;
-			clock-names = "otg";
-			phys = <&usbphy>;
-			phy-names = "usb2-phy";
-			status = "disabled";
-		};
-
 		mop: mop@7c500000 {
 			compatible = "brcm,bcm2712-mop";
 			reg = <0x7c500000 0x20>;
@@ -1145,6 +1132,19 @@
 			reg = <0x10 0x00400018  0x0 0x18>;
 		};
 
+		usb: usb@480000 {
+			compatible = "brcm,bcm2835-usb";
+			reg = <0x10 0x00480000 0x0 0x10000>;
+			interrupts = <GIC_SPI 73 IRQ_TYPE_LEVEL_HIGH>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			clocks = <&clk_usb>;
+			clock-names = "otg";
+			phys = <&usbphy>;
+			phy-names = "usb2-phy";
+			status = "disabled";
+		};
+
 		rpivid: codec@800000 {
 			compatible = "raspberrypi,rpivid-vid-decoder";
 			reg = <0x10 0x00800000  0x0 0x10000>, /* HEVC */
